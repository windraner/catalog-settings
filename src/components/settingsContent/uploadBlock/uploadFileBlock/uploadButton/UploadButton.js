import React, { Component } from 'react';
import DisabledUploadButton from './disabledUploadButton/DisabledUploadButton';
import UploadButtonRemove from './uploadButtonRemove/UploadButtonRemove';

import './uploadButton.css';

export default class UploadButton extends Component {
  changeHandler = (e) => {
    this.props.inputHandler('file', e.target.files[0]);
  }

  render() {
    const { file, fileUrl, inputHandler } = this.props;

    if(file) {
      return (
        <UploadButtonRemove
          inputHandler={inputHandler}
          text={file.name}
        />
      );
    }

    if(fileUrl) {
      return (
        <DisabledUploadButton />
      );
    }

    return (
      <label htmlFor="upload_doc" className="catalog-settings-content__upload-button">
        <span className="catalog-settings-content__upload-button-icon"></span>
        <span>Upload file</span>
        <input
          accept=".csv"
          onChange={this.changeHandler}
          id="upload_doc"
          type="file"
          className="catalog-settings-content__upload-input"
        />
      </label>
    );
  }
}
