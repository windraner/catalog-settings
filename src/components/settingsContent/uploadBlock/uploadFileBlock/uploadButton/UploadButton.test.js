import React from 'react';
import UploadButton from './UploadButton';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<UploadButton />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<UploadButton inputHandler={function() {}} fileUrl={''} file={''} flags={{}} />);
    const button = wrapper.find('.catalog-settings-content__upload-button');
    expect(button.text()).toBe('Upload file');
  });
  it('Check displaing remove button', () => {
    const wrapper = mount(<UploadButton inputHandler={function() {}} fileUrl={'fileUrl'} file={{name: 'test'}} flags={{}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Check displaing enabled button', () => {
    const wrapper = mount(<UploadButton inputHandler={function() {}} fileUrl={'fileUrl'} file={''} flags={{}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
