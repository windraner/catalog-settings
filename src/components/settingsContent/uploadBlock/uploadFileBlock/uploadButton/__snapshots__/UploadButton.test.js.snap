// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadButton /> Check displaing enabled button 1`] = `
<UploadButton
  file=""
  fileUrl="fileUrl"
  flags={Object {}}
  inputHandler={[Function]}
>
  <DisabledUploadButton>
    <div
      className="catalog-settings-content__upload-button catalog-settings-content__upload-button_disabled"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <ToolTip
        isShow={false}
        text="Remove URL to manually add CSV file"
      />
      <span
        className="catalog-settings-content__upload-button-icon"
      />
      <span>
        Upload file
      </span>
    </div>
  </DisabledUploadButton>
</UploadButton>
`;

exports[`<UploadButton /> Check displaing remove button 1`] = `
<UploadButton
  file={
    Object {
      "name": "test",
    }
  }
  fileUrl="fileUrl"
  flags={Object {}}
  inputHandler={[Function]}
>
  <UploadButtonRemove
    inputHandler={[Function]}
    text="test"
  >
    <div
      className="catalog-settings-content__remove-upload-button"
    >
      <span>
        test
      </span>
      <span
        className="catalog-settings-content__upload-button-remove-icon"
        onClick={[Function]}
      />
    </div>
  </UploadButtonRemove>
</UploadButton>
`;
